fil-proofs-param/src/bin/paramcache.rs

```diff
@@ -22,14 +24,14 @@ use storage_proofs_core::{
 };
 use storage_proofs_porep::stacked::{StackedCircuit, StackedCompound, StackedDrg};
 use storage_proofs_post::fallback::{FallbackPoSt, FallbackPoStCircuit, FallbackPoStCompound};
-use storage_proofs_update::constants::TreeRHasher;
-use storage_proofs_update::{
-    circuit::EmptySectorUpdateCircuit, compound::EmptySectorUpdateCompound, EmptySectorUpdate,
-    PublicParams,
-};
+use storage_proofs_update::{constants::TreeRHasher, EmptySectorUpdateCompound};
 use structopt::StructOpt;
 
-fn cache_porep_params<Tree: 'static + MerkleTreeTrait>(porep_config: PoRepConfig) {
+fn cache_porep_params<Tree>(porep_config: PoRepConfig)
+where
+    Tree: 'static + MerkleTreeTrait,
+    <Tree::Hasher as Hasher>::Domain: Domain<Field = Fr>,
```

Comment:
Note: this trait bound exists because Groth16 params can only be generated using the `Fr` field.